pkg/sidecar/appclone.go

```diff
@@ -171,6 +241,20 @@ func xtrabackupPreperData() error {
 }
 
 func deleteLostFound() error {
-	path := fmt.Sprintf("%s/lost+found", dataDir)
-	return os.RemoveAll(path)
+	lfPath := fmt.Sprintf("%s/lost+found", dataDir)
+	return os.RemoveAll(lfPath)
+}
+
+func cleanDataDir() {
```

Comment:
This was done to avoid leaving the `dataDir` in an unknown or partially restored state. The case we ran into was a large backup (hundreds of GBs) being interrupted midstream due to a network blip. On subsequent pod re-starts, mysql would continually fail to start as the PVC had only a partial backup.